### DISCUSSION TITLE
GeForce 210 Opencore - is it possible ?

### DISCUSSION THREAD

**Discussion by sektorcoder** [chris1111/Legacy-Video-patch#17] [Q&A]
First of all thank you for your work, it allowed me to easily test a Mac with legacy Nvidia without much hassle and am pretty happy with it.
I started with a KVM - Clover in GPU passthrough mode (Nvidia GeForce 210) using a linux host and applied your work, everything worked like a charm.
Then I wanted to transfer the same functionality in an opencore AMD 3900x - Aorus Elite X570 bare metal system.
While everything worked fine (installation, settings, system booted, even patching the Nvidia card according to Opencore documentation) it failed after applying the legacy-video-patch.

The most usual error was this:

```panic(cpu 0 caller 0xffffff800e6459fa): Kernel trap at 0xffffff800ebb7974, type 14=page fault, registers:
CR0: 0x000000008001003b, CR2: 0x00000000a790c600, CR3: 0x000000001cd7f000, CR4: 0x00000000003406e0
RAX: 0x0000000021000000, RBX: 0xffffff80a4776920, RCX: 0x0000000021000001, RDX: 0xffffff869bedb000
RSP: 0xffffff874f673b90, RBP: 0xffffff874f673b90, RSI: 0x00000000a790c600, RDI: 0xffffff80a4776920
R8:  0x0000000000000001, R9:  0x0000000000000002, R10: 0x0000000000000002, R11: 0xffffff80a48b81c0
R12: 0xffffff869bedb000, R13: 0x0000000000000000, R14: 0x00000000a790c600, R15: 0xffffff80a51d3080
RFL: 0x0000000000010282, RIP: 0xffffff800ebb7974, CS:  0x0000000000000008, SS:  0x0000000000000010
Fault CR2: 0x00000000a790c600, Error code: 0x0000000000000000, Fault CPU: 0x0, PL: 0, VF: 1

Backtrace (CPU 0), Frame : Return Address
0xffffff874f6735f0 : 0xffffff800e51961d mach_kernel : _handle_debugger_trap + 0x49d
0xffffff874f673640 : 0xffffff800e653ac5 mach_kernel : _kdp_i386_trap + 0x155
0xffffff874f673680 : 0xffffff800e64564e mach_kernel : _kernel_trap + 0x4ee
0xffffff874f6736d0 : 0xffffff800e4bfa40 mach_kernel : _return_from_trap + 0xe0
0xffffff874f6736f0 : 0xffffff800e518ce7 mach_kernel : _DebuggerTrapWithState + 0x17
0xffffff874f6737f0 : 0xffffff800e5190d7 mach_kernel : _panic_trap_to_debugger + 0x227
0xffffff874f673840 : 0xffffff800ecc00bc mach_kernel : _panic + 0x54
0xffffff874f6738b0 : 0xffffff800e6459fa mach_kernel : _sync_iss_to_iks + 0x2aa
0xffffff874f673a30 : 0xffffff800e6456f8 mach_kernel : _kernel_trap + 0x598
0xffffff874f673a80 : 0xffffff800e4bfa40 mach_kernel : _return_from_trap + 0xe0
0xffffff874f673aa0 : 0xffffff800ebb7974 mach_kernel : __ZNK12OSSymbolPool10findSymbolEPKc + 0x4
0xffffff874f673b90 : 0xffffff800ebb7e4f mach_kernel : __ZN8OSSymbol11withCStringEPKc + 0x2f
0xffffff874f673bc0 : 0xffffff800eb97dd8 mach_kernel : __ZN12OSDictionary9setObjectEPK8OSStringPK15OSMetaClassBase + 0x18
0xffffff874f673bf0 : 0xffffff7f90ab6b6e com.apple.iokit.IOSurface : __ZN13IOSurfaceRoot15addMemoryRegionEP21IOSurfaceMemoryRegion + 0x42
0xffffff874f673c20 : 0xffffff7f92aac52d com.apple.GeForceTesla : __ZN20IONVAcceleratorTesla5startEP9IOService + 0xc7
0xffffff874f673d10 : 0xffffff7f92acb188 com.apple.GeForceTesla : __ZN8NVKernel5startEP9IOService + 0xc6
0xffffff874f673de0 : 0xffffff800ec05bc6 mach_kernel : __ZN9IOService14startCandidateEPS_ + 0xf6
0xffffff874f673e40 : 0xffffff800ec05701 mach_kernel : __ZN9IOService15probeCandidatesEP12OSOrderedSet + 0xad1
0xffffff874f673ef0 : 0xffffff800ec04a9e mach_kernel : __ZN9IOService14doServiceMatchEj + 0x2de
0xffffff874f673f50 : 0xffffff800ec07256 mach_kernel : __ZN15_IOConfigThread4mainEPvi + 0x186
0xffffff874f673fa0 : 0xffffff800e4bf13e mach_kernel : _call_continuation + 0x2e
      Kernel Extensions in backtrace:
         com.apple.iokit.IOSurface(269.11)[8F3433B2-7A63-38F2-9B09-585BA47F9BB6]@0xffffff7f90aa9000->0xffffff7f90ad7fff
         com.apple.GeForceTesla(10.0.4)[426B2568-8B0A-372B-BB33-B421D942837A]@0xffffff7f92aa7000->0xffffff7f92b9efff
            dependency: com.apple.iokit.IOPCIFamily(2.9)[0F12857B-4B0E-3060-9519-2B65B17CC60C]@0xffffff7f8ef12000
            dependency: com.apple.iokit.IONDRVSupport(519.15)[B419F958-11B8-3F7D-A31B-A72166B6E234]@0xffffff7f9257e000
            dependency: com.apple.iokit.IOGraphicsFamily(519.15)[D5F2A20D-CAB0-33B2-91B9-E8755DFC34CB]@0xffffff7f92465000
            dependency: com.apple.nvidia.classic.NVDAResmanTesla(10.0.4)[F8258F0F-61AD-30F4-BDD5-B18CCAAE535A]@0xffffff7f92595000

BSD process name corresponding to current thread: kernel_task
Boot args: -v blah,blah,blah,blah,<various tests> keepsyms=1 debug=0x100 npci=0x2000 chunklist-security-epoch=0 -chunklist-no-rev2-dev chunklist-security-epoch=0 -chunklist-no-rev2-dev

Mac OS version:
19H114

Kernel version:
Darwin Kernel Version 19.6.0: Tue Nov 10 00:10:30 PST 2020; root:xnu-6153.141.10~1/RELEASE_X86_64
Kernel UUID: A477D537-F87E-3978-8D7F-A84B7C02E62E
Kernel slide:     0x000000000e200000
Kernel text base: 0xffffff800e400000
__HIB  text base: 0xffffff800e300000
System model name: MacPro7,1 (Mac-27AD2F918AE68F61)
System shutdown begun: NO
Panic diags file available: NO (0xe00002bc)

System uptime in nanoseconds: 14129578039
```

Do you have any tip on what could be wrong ? Are there kext dependency / functionality hell here or you suspect something more simple ?
I am totally new to this and I spent two days searching for similar errors and multiple opencore config tests without success.
If you suspect this is something complex a general direction to start learning on the matter would also help me.

Kind regards

**Comment by chris1111** [comment]
I don't know because I use Clover for Legacy video patch so I can not really help you
see my EFI here https://github.com/chris1111/Legacy-Video-patch/files/5720249/EFI.Legacy.Video.Patch.Clover.r5127.zip

**Reply by sektorcoder** [reply]
> I don't know because I use Clover for Legacy video patch so I can not really help you
> see my EFI here https://github.com/chris1111/Legacy-Video-patch/files/5720249/EFI.Legacy.Video.Patch.Clover.r5127.zip

Ok, thanks for the feedback, I will check your settings.

**Reply by chris1111** [reply]
> > I don't know because I use Clover for Legacy video patch so I can not really help you
> > see my EFI here https://github.com/chris1111/Legacy-Video-patch/files/5720249/EFI.Legacy.Video.Patch.Clover.r5127.zip
> 
> Ok, thanks for the feedback, I will check your settings.

take me inform of your progress Good luck 😀

**Reply by w32u** [reply]
I have the same kernel panic on OpenCore FX-4350 + XFX 8800 GS build, but my SMBIOS is iMac14,2. I've tried to boot up with -no_compat_check and amfi_get_out_of_my_way=1, but it eventually failed. Don't know how to fix it right now.
Steps to reproduce:
1. Install 10.15.7 Catalina with OpenCore and AMD 15h CPU patches according to Dortania guide
2. Install last update
3. Patch legacy NVIDIA card in config according to Dortania guide
4. Install Legacy video patch
5. Get kernel panic after installation

The thing is that Clover isn't supported anymore for AMD CPUs by AMDOSX community
I've also made post on Reddit about this problem with the picture of panic: https://www.reddit.com/r/hackintosh/comments/mi3jbc/kernel_panic_after_applying_legacy_video_patch_to/
